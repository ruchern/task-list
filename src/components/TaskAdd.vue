<template>
  <div class="box">
    <div class="field">
      <label for="task" class="label">Add Task</label>
      <input type="text" id="task" class="input" :class="{ 'is-success': name.length > 0 }"
             placeholder="e.g. Remember to write tests for your project!" v-model="name" @keyup.enter="addTask">
      <p class="content has-text-danger has-text-weight-bold">{{ message }}</p>
    </div>
  </div>
</template>

<script>
export default {
  name: 'TaskAdd',
  data () {
    return {
      name: '',
      message: null
    }
  },
  methods: {
    addTask () {
      const task = {
        name: this.name,
        completed: false
      }

      if (this.name.length > 0) {
        this.$store.commit('addTask', task)
      } else {
        this.message = 'Your task cannot be empty.'
      }

      this.name = ''
    }
  }
}
</script>

<style lang="scss" scoped>
.box {
  width: 100%;
}
</style>
