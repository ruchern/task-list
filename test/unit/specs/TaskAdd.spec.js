import { createLocalVue, shallow } from '@vue/test-utils'
import Vuex from 'vuex'
import TaskAdd from '@/components/TaskAdd'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('TaskAdd', () => {
  let wrapper

  let actions
  let store

  beforeEach(() => {
    actions = {
      addTask: jest.fn()
    }

    store = new Vuex.Store({
      state: {},
      actions
    })

    wrapper = shallow(TaskAdd, {
      store, localVue
    })
  })

  test('message should be null when task name is not empty', () => {
    const input = wrapper.find('input')
    input.element.value = 'Task 1'
    input.trigger('keyup')

    expect(wrapper.vm.message).toBeNull()
  })

  test('add task when button is clicked', () => {
    const input = wrapper.find('input')
    input.element.value = 'Task 1'
    input.trigger('input')

    wrapper.find('button').trigger('click')

    expect(actions.addTask).toHaveBeenCalled()
  })

  test('should display a message if task name is empty when button is clicked', () => {
    const input = wrapper.find('input')
    input.element.value = ''
    input.trigger('input')

    wrapper.find('button').trigger('click')

    expect(wrapper.vm.message).toBe('Your task cannot be empty.')
  })
})
