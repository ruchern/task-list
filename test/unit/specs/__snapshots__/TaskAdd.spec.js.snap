// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskAdd should render the component 1`] = `
<div
  class="box"
>
  <div
    class="field"
  >
    <label
      class="label"
      for="task"
    >
      Add Task
    </label>
     
    <input
      class="input"
      id="task"
      placeholder="e.g. Remember to write tests for your project!"
      type="text"
    />
     
    <p
      class="content has-text-danger has-text-weight-bold"
    >
      
    </p>
  </div>
</div>
`;
